File: lb-processor/src/main/java/com/github/rooneyandshadows/lightbulb/annotation_processors/annotations/FragmentConfiguration.java
package com.github.rooneyandshadows.lightbulb.annotation_processors.annotations;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.SOURCE)
@Target(ElementType.TYPE)
public @interface FragmentConfiguration {
    String layoutName() default "";

    boolean isMainScreenFragment() default true;

    boolean hasLeftDrawer() default false;

    boolean hasOptionsMenu() default false;
}

File: src/main/java/com/github/rooneyandshadows/lightbulb/annotation_processors/annotations/FragmentConfiguration.kt
package com.github.rooneyandshadows.lightbulb.annotation_processors.annotations

@Retention(AnnotationRetention.SOURCE)
@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS)
annotation class FragmentConfiguration(
    val layoutName: String = "",
    val isMainScreenFragment: Boolean = true,
    val hasLeftDrawer: Boolean = false,
    val hasOptionsMenu: Boolean = false
)